linear_algebra/src/gaussian_elimination_pivoting/gaussian_elimination_pivoting.py

```diff
@@ -0,0 +1,93 @@
+import sys
+import time
+
+import numpy as np
+
+matrixab = np.loadtxt("matrix.txt")
+B = np.copy(matrixab[:, matrixab.shape[1] - 1])
+
+
+def foo(matrix):
```

Comment:
Please.  Come up with a better name for this function and add a type hint and maybe even some doctests.

According to `CONTRIBUTING.md` algorithmic functions should not `print()`.

Reply:
Function is renamed to solve_linear_system, and type hints are added for parameters and return values. The doctest provides an example of usage and asserts the correctness of the solution. I've also removed the print statements and replaced the exit call with a sys.exit exception to allow for better error handling.